File: lifecyklelog/src/main/java/com/chesire/lifecyklelog/LogHandler.java
package com.chesire.lifecyklelog;

import android.os.Bundle;

import androidx.annotation.Nullable;

/**
 * Handles logging out lifecycle methods.
 */
public interface LogHandler {
    /**
     * Executed when a lifecycle method requires logging.
     *
     * @param clazz          string representation of the class that the lifecycle event occurred on.
     * @param lifecycleEvent string representation of the lifecycle event that occurred.
     * @param bundle         the bundle passed into the lifecycle event, or null if not available.
     */
    void logLifecycleMethod(String clazz, String lifecycleEvent, @Nullable Bundle bundle);
}

File: lifecyklelog/src/main/java/com/chesire/lifecyklelog/LogHandler.kt
package com.chesire.lifecyklelog

import android.os.Bundle

/**
 * Handles logging out lifecycle methods.
 */
fun interface LogHandler {
    /**
     * Executed when a lifecycle method requires logging.
     *
     * [clazz] is the string representation of the class that the lifecycle event occurred on.
     * [lifecycleEvent] is the string representation of the lifecycle event that occurred.
     * [bundle] the bundle passed into the lifecycle event, or null if not available.
     */
    fun logLifecycleMethod(clazz: String, lifecycleEvent: String, bundle: Bundle?)
}
